src/plugins/github/referenceDetector.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+
+import dedent from "../../util/dedent";
+import {type NodeAddressT} from "../../core/graph";
+import {MappedReferenceDetector, type URL} from "../../core/references";
+import {RelationalView} from "./relationalView";
+
+/**
+ * Builds a GithubReferenceDetector using multiple RelationalView.
+ * As RelationalView should only be used for one repository at a time, you will
+ * commonly want to compose several of them into one GithubReferenceDetector.
+ *
+ * Note: duplicates are normally expected. However for any URL, the corresponding
+ * NodeAddressT should be the same, or we'll throw an error.
+ */
+export function fromRelationalViews(
+  views: $ReadOnlyArray<RelationalView>
+): GithubReferenceDetector {
+  const map: Map<URL, NodeAddressT> = new Map();
+  for (const view of views) {
+    for (const [url, addr] of view.urlReferenceMap().entries()) {
+      const existing = map.get(url);
+      if (existing && existing != addr) {
+        throw new Error(dedent`\
+          An entry for ${url} already existed, but with a different NodeAddressT.
+          This is probably a bug with SourceCred. Please report it on GitHub.
+          Old: ${existing}
```

Comment:
if you want the address to be readable for humans, it's best to use this method which pretty prints so we dont put nul bytes in the error message.

https://github.com/sourcecred/sourcecred/blob/c1f123f3522e8cab4c83beb74e775119fd93a049/src/core/address.js#L43-L48

Reply:
Applied for both PRs